docs/recipes.rst

```diff
@@ -189,10 +189,9 @@ hook:
                 # Simple build, do nothing.
                 return
 
-            if extracted:
-                # A list of groups were passed in, use them
-                for group in extracted:
-                    self.groups.add(group)
+            # An iterable of groups were passed in, use it
+            if isinstance(extracted, (list, set, tuple, models.QuerySet)):
```

Comment:
Any iterable?

```suggestion
            if isinstance(extracted, collections.abc.Iterable):
```

Reply:
`str` is iterable 😬

Reply:
Why not:
```python
if extracted:
    # An iterable of groups were passed in, use it
    self.groups.add(*extracted) # bulk addition
```

My concern is that passing a value that’s not a (list, set, tuple, models.QuerySet) would be silently ignored. If groups is passed, it must be an iterable that yields group objects, otherwise fail.

Reply:
I'm perfectly fine with that, this was my first proposal (see upper discussion). I just thought it might be too harsh for some people, but I also prefer to fail earlier!